.github/workflows/ci.yml

```diff
@@ -0,0 +1,408 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - main
+      - develop
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: 1
+
+permissions:
+  checks: write
+
+jobs:
+  mc-tests:
+    runs-on: [self-hosted, Linux, large]
+    container: mobilecoin/rust-sgx-base:latest
+
+    strategy:
+      matrix:
+        num_runners: [2]
+        runner_index: [1, 2]
+      # Run each shard to completion.
+      fail-fast: false
+    env:
+      NUM_RUNNERS: ${{ matrix.num_runners }}
+      RUNNER_INDEX: ${{ matrix.runner_index }}
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+      - name: List packages to test
+        run: |
+          cargo test -p 2>&1 | \
+            awk "/mc-/ { print \"-p \" \$1 }" | \
```

Comment:
What happens if we ever have tests in a package that doesn't start with `mc-`? Right now we have `libmobilecoin` and it doesn't have tests, but if we did add them they would not run, so I am hoping there is a more robust way to do what you're doing here.

Reply:
Fair point. I wanted to prune the first few lines of the error output, and match only on package names. We could do that by checking for leading whitespace, but that seems hackier.

FWIW the output starts like this:
```
error: "--package <SPEC>" requires a SPEC format value, which can be any package ID specifier in the dependency graph.
Run `cargo help pkgid` for more information about SPEC format.

Possible packages/workspace members:
    mc-account-keys
```

Reply:
`cargo metadata --no-deps | jq -r '.packages[].name'` might be a good alternative